core/core-api/src/main/java/org/kontinuity/catapult/core/api/ProjectileBuilder.java

```diff
@@ -1,189 +1,125 @@
-        return this;
-    }
-    
-    /**
-     * Sets the GitHub access token we have obtained from the user as part of
-     * the OAuth process. Required.
-     *
-     * @param gitHubAccessToken
-     * @return This builder
-     */
-    public ProjectileBuilder gitHubAccessToken(final String gitHubAccessToken) {
-        this.gitHubAccessToken = gitHubAccessToken;
-        return this;
-    }
+   void build(ProjectileBuilder builder) {
+      ProjectileBuilder.checkSpecified("gitHubAccessToken", this.gitHubAccessToken);
+      // Default the openshiftProjectName if need be
+      try {
+         ProjectileBuilder.checkSpecified("openshiftProjectName", this.openShiftProjectName);
+      } catch (final IllegalStateException ise) {
+         this.openShiftProjectName(builder.createDefaultProjectName());
+      }
+   }
 
-   /**
-    * Sets the path to file that contains the template to apply on the
-    * OpenShift project. Required.
-    *
-    * @param pipelineTemplatePath
-    * @return This builder
-    */
-   public ProjectileBuilder pipelineTemplatePath(final String pipelineTemplatePath) {
-      this.pipelineTemplatePath = pipelineTemplatePath;
-      return this;
+   String createDefaultProjectName() {
+      throw new IllegalStateException("needs to be called on specific type");
    }
 
    /**
     * Sets the name of the OpenShift project to create. By default, the name is derived from
     * the GitHub repository to fork. Optional.
+    *
     * @param openShiftProjectName
     * @return This builder
     */
    public ProjectileBuilder openShiftProjectName(final String openShiftProjectName) {
-      this.openShiftProjectName = openShiftProjectName;;
+      this.openShiftProjectName = openShiftProjectName;
       return this;
    }
 
    /**
-    * Sets Git ref to use. Required
+    * Sets the GitHub access token we have obtained from the user as part of
+    * the OAuth process. Required.
     *
-    * @param gitRef
+    * @param gitHubAccessToken
     * @return This builder
     */
-   public ProjectileBuilder gitRef(final String gitRef) {
-      this.gitRef = gitRef;
+   public ProjectileBuilder gitHubAccessToken(final String gitHubAccessToken) {
+      this.gitHubAccessToken = gitHubAccessToken;
       return this;
    }
 
-	/**
-     * @return source GitHub repository name in form "owner/repoName".
-     */
-    public String getSourceGitHubRepo() {
-        return this.sourceGitHubRepo;
-    }
-
-    /**
-     * @return the GitHub access token we have obtained from the user as part of
-     * the OAuth process
-     */
-    public String getGitHubAccessToken() {
-        return this.gitHubAccessToken;
-    }
-
    /**
-    * @return The name to use in creating the new OpenShift project
+    * @return the GitHub access token we have obtained from the user as part of
+    * the OAuth process
     */
-   public String getOpenShiftProjectName() { return openShiftProjectName;  }
+   public String getGitHubAccessToken() {
```

Comment:
I think we should get rid of these getter methods in the Builder. They are already available in the built Projectile